nan.h

```diff
@@ -1122,51 +1136,53 @@ widenString(std::vector<uint16_t> *ws, const uint8_t *s, int l) {
   NewOneByteString(const uint8_t * value, int length = -1) {
     std::vector<uint16_t> wideString;  // NOLINT(build/include_what_you_use)
     imp::widenString(&wideString, value, length);
-    return imp::Factory<v8::String>::return_t(v8::String::New(
-        &wideString.front(), static_cast<int>(wideString.size())));
+    return v8::String::New(&wideString.front(),
```

Comment:
Using .front() when the string is empty is UB.  You could use .data() instead, that has well-defined behavior for the .empty() case (pointer to a `\0` byte as of C++11, something I learned earlier this week.)